src/Node/ForNode.php

```diff
@@ -25,7 +25,7 @@
 #[YieldReady]
 class ForNode extends Node
 {
-    public function __construct(AssignContextVariable $keyTarget, AssignContextVariable $valueTarget, AbstractExpression $seq, ?AbstractExpression $ifexpr, Node $body, ?Node $else, int $lineno)
+    public function __construct(AssignContextVariable $keyTarget, AssignContextVariable $valueTarget, AbstractExpression $seq, ?AbstractExpression $ifexpr, Node $body, ?ForElseNode $else, int $lineno)
```

Comment:
Restricting the type is a BC break. You should introduce the new class in 3.x with a deprecation instead (where passing something else than a `ForElseNode` or `null` would trigger a deprecation and wrap the node in a ForElseNode, using the line of that node as fallback line number)

Reply:
Let's start with the v3 PR.